Entitas/Entitas/Entity.cs

```diff
@@ -6,37 +6,18 @@ namespace Entitas {
 
     /// Use pool.CreateEntity() to create a new entity and pool.DestroyEntity() to destroy it.
     /// You can add, replace and remove IComponent to an entity.
-    public partial class Entity {
+    public partial class Entity : IEntity {
 
-        /// Occurs when a component gets added. All event handlers will be removed when the entity gets destroyed by the pool.
         public event EntityChanged OnComponentAdded;
-
-        /// Occurs when a component gets removed. All event handlers will be removed when the entity gets destroyed by the pool.
         public event EntityChanged OnComponentRemoved;
-
-        /// Occurs when a component gets replaced. All event handlers will be removed when the entity gets destroyed by the pool.
         public event ComponentReplaced OnComponentReplaced;
 
-        public delegate void EntityChanged(Entity entity, int index, IComponent component);
-        public delegate void ComponentReplaced(Entity entity, int index, IComponent previousComponent, IComponent newComponent);
-
-        /// The total amount of components an entity can possibly have.
         public int totalComponents { get { return _totalComponents; } }
-
-        /// Each entity has its own unique creationIndex which will be set by the pool when you create the entity.
         public int creationIndex { get { return _creationIndex; } }
-
-        /// The pool manages the state of an entity. Active entities are enabled, destroyed entities are disabled.
         public bool isEnabled { get { return _isEnabled; } }
 
-        /// componentPools is set by the pool which created the entity and is used to reuse removed components.
-        /// Removed components will be pushed to the componentPool.
-        /// Use entity.CreateComponent(index, type) to get a new or reusable component from the componentPool.
-        /// Use entity.GetComponentPool(index) to get a componentPool for a specific component index.
         public Stack<IComponent>[] componentPools { get { return _componentPools; } }
 
-        /// The poolMetaData is set by the pool which created the entity and contains information about the pool.
-        /// It's used to provide better error messages.
```

Comment:
I move all these comments to the interface so they can be maintained in just one location.